### PR TITLE
feat: add tag resources

### PR DESCRIPTION
**Pull request by Fuochi**
- add GetTag
- add DeleteTag
- add Tag methods for Prowlarr (not very useful)
- edit UpdateTag to be aligned with api docs

### CODE REVIEW

File: prowlarr/tag.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+// UpdateTag updates the label for a tag.
+func (p *Prowlarr) UpdateTag(tagID int, label string) (int, error) {
+	return p.UpdateTagContext(context.Background(), tagID, label)
+}
+
+// UpdateTagContext updates the label for a tag.
+func (p *Prowlarr) UpdateTagContext(ctx context.Context, tagID int, label string) (int, error) {
+	body, err := json.Marshal(&starr.Tag{Label: label, ID: tagID})
```
Comment (davidnewhall) [inline prowlarr/tag.go:56]: dope.

File: prowlarr/tag.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	var tag starr.Tag
+	if err = p.PostInto(ctx, "v1/tag", nil, body, &tag); err != nil {
+		return tag.ID, fmt.Errorf("api.Post(tag): %w", err)
+	}
+
+	return tag.ID, nil
+}
+
+// UpdateTag updates the label for a tag.
+func (p *Prowlarr) UpdateTag(tagID int, label string) (int, error) {
```
Comment (davidnewhall) [inline prowlarr/tag.go:50]: Is Prowlarr the only one with update?

**Review by davidnewhall — APPROVED** [review]
This is beautiful. Thanks!

**Comment by davidnewhall** [comment]
I'll cut a new release in the next couple weeks. Got a few other changes that are pending testing. Feel free to add anything else you need.

File: prowlarr/tag.go (hunk 2 above)
Comment (Fuochi) [reply prowlarr/tag.go:50]: no, the others already had update, I just changed it to add the ID. It works even without it, but in all api docs but sonarr's the ID is in the definition (e.g. https://radarr.video/docs/api/#/Tag/put-tag-id)

**Comment by Fuochi** [comment]
> 

Thanks! I really like this work, I'll try few things with the tags. Once I finished, I'll probably contribute more! But it could take me a while.

btw, do you have in plan to write any unit tests?

**Comment by davidnewhall** [comment]
I'd love tests, I suppose I could start a few and get a base going at least. I'm just lazy on this one. :)
I did add some to the new submodule I wrote in another PR... so I'm trying!

 Thanks again!

File: prowlarr/tag.go (hunk 2 above)
Comment (davidnewhall) [reply prowlarr/tag.go:50]: Gotcha, cool. I think something I wrote uses update tag, and it's been around a while so I never made it back there to the rest of the methods.

**Comment by davidnewhall** [comment]
fwiw, I started some tests. If you have feedback on where I'm going with those, I'm all ears.

https://github.com/golift/starr/pull/33

**Pull request merged**